tests/yast2_gui/yast2_firewall_set_interface.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: YaST2 Firewall UI test checks verious configurations and settings of firewall
+# Make sure yast2 firewall can opened properly. Configurations can be changed and written correctly.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use network_utils 'iface';
+use YaST::Module;
+use version_utils "is_sle";
+
+sub verify_interface {
+    my (%args) = @_;
+    validate_script_output("firewall-cmd --list-interfaces --zone=$args{zone}", sub { m/$args{device}/ }, proceed_on_failure => 1);
+}
+
+sub change_interface_zone {
+    my (%args) = @_;
+    $testapi::distri->get_firewall()->select_interfaces_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_interface_zone($args{device}, $args{zone});
+    save_screenshot;
+}
+
+sub run {
+    my $self = shift;
+    my $iface = iface();
+
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    change_interface_zone(device => $iface, zone => 'public');
```

Comment:
same comment than below, let's not use methods, and have the whole code here, it is simple enough, and then we avoid to contributors to try to create a library with those or something more strange.